src/uuid/rfc4122.cr

```diff
@@ -0,0 +1,85 @@
+# Support for RFC 4122 UUID variant.
+struct UUID
+  # RFC 4122 UUID variant versions.
+  enum Version
+    # Unknown version.
+    Unknown = 0
+
+    # Version 1 - date-time and MAC address.
+    V1 = 1
+
+    # Version 2 - DCE security.
+    V2 = 2
+
+    # Version 3 - MD5 hash and namespace.
+    V3 = 3
+
+    # Version 4 - random.
+    V4 = 4
+
+    # Version 5 - SHA1 hash and namespace.
+    V5 = 5
+  end
+
+  # Generates RFC 4122 UUID `variant` with specified `version`.
+  def initialize(version : Version)
+    case version
+    when Version::V4
+      @bytes.to_unsafe.copy_from SecureRandom.random_bytes(16).pointer(16), 16
```

Comment:
We can avoid a copy and directly fill `@bytes`, which also avoids a HEAP allocation :

```crystal
@bytes = uninitialized StaticArray(UInt8, 16)
SecureRandom.random_bytes(@bytes.to_slice)
```

Also, maybe we could per version factories, for a nicer API?
```crystal
def self.random
  bytes = uninitialized StaticArray(UInt8, 16)
  SecureRandom.random_bytes(bytes.to_slice)

  uuid = new(bytes)
  uuid.variant = Variant::RFC4122
  uuid.version = Version::V4
  uuid
end
```